wp-includes/widgets/class-wp-widget-media.php

```diff
@@ -234,6 +235,45 @@ class="media-widget-instance-property"
 	}
 
 	/**
+	 * Deletes the Widget when the attachment file is deleted.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 *
+	 * @param int $post_id Attachment ID.
+	 */
+	public function delete_attachment_data( $post_id ) {
+		$settings      = $this->get_settings();
+		$attachment_id = empty( $settings[ $this->number ]['attachment_id'] ) ? 0 : $settings[ $this->number ]['attachment_id'];
```

Comment:
As done in `display_media_state `, this will need to loop over all `$settings` not just `$settings[ $this->number ]`.